ChaosMod/Effects/db/Player/PlayerZoomZoomCam.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+    Effect by Moxi, based on code from DrUnderscore & ProfessorBiddle
+*/
+
+#include <stdafx.h>
+static Camera zoomCamera = 0;
+float camZoom = 80.f;
+const float camZoomRate = 0.6f;
+int n = 0;
+
+static void UpdateCamera()
+{
+    auto coord = CAM::GET_GAMEPLAY_CAM_COORD();
+    auto rot = CAM::GET_GAMEPLAY_CAM_ROT(2);
+    CAM::SET_CAM_PARAMS(zoomCamera, coord.x, coord.y, coord.z, rot.x, rot.y, rot.z, camZoom, 0, 1, 1, 2);
+}
+
+static void OnStart()
+{
+    zoomCamera = CAM::CREATE_CAM("DEFAULT_SCRIPTED_CAMERA", 1);
+    CAM::RENDER_SCRIPT_CAMS(true, true, 10, 1, 1, 1);
+}
+
+static void OnTick()
+{
+    static DWORD64 lastTick = 0;
+    DWORD64 curTick = GET_GAME_TIMER();
+
+    if (curTick > lastTick + 5)
+    {
+        lastTick = curTick;
+
+        if (camZoom < 11)
+        {
+            n = 0;
+        }
+        if (camZoom > 119)
+        {
+            n = 1;
+        }
+        if (n == 0)
+        {
+            camZoom += camZoomRate;
+            CAM::SET_CAM_ACTIVE(zoomCamera, true);
+            UpdateCamera();
+        }
+        if (n == 1)
+        {
+            camZoom -= camZoomRate;
+            CAM::SET_CAM_ACTIVE(zoomCamera, true);
+            UpdateCamera();
+        }
+    }
+}
+
+static void OnStop()
+{
+    CAM::SET_CAM_ACTIVE(zoomCamera, false);
+    CAM::RENDER_SCRIPT_CAMS(false, true, 700, 1, 1, 1);
+    CAM::DESTROY_CAM(zoomCamera, true);
+    zoomCamera = 0;
+}
+
+static RegisterEffect registerEffect(EFFECT_PLAYER_ZOOMZOOM_CAM, OnStart, OnStop, OnTick, EffectInfo
+    {
+        .Name = "Zoom Zoom Cam",
+        .Id = "player_zoomzoom_cam",
+        .IsTimed = true,
+        .IsShortDuration = true,
+        .IncompatibleWith = { EFFECT_FLIP_CAMERA, EFFECT_PLAYER_GTA_2, EFFECT_PLAYER_QUAKE_FOV, EFFECT_PLAYER_BINOCULARS }
```

Comment:
this effect has to be added to those other effects too :)

Reply:
Really? Cant't you just check both? Weird rule imo

Reply:
I've tested ingame and it's working fine without it I believe. If I start Binoculars then Zoom Zoom, or the other way around it always disable the first one before doing the second.
Of course I can still add the exceptions to the others effects if you want me to, but why are they required? I don't know what I'm missing.

Reply:
I think it's meant more for code cleanliness than function. Maybe a list of effect groups that are incompatible with each other in a separate file might be good eventually

Reply:
Okay I understand. I've added it to the other effects. I also added Binoculars to Flip, GTA2 and Quake because it was missing. I hope it's okay